### PR TITLE
Improve compatibility with Torch7

### PR DESCRIPTION
**Pull request by mvitez**
Some torch7 networks did not run. These are some fixes as described here:

https://discuss.pytorch.org/t/convert-import-torch-model-to-pytorch/37/7

### CODE REVIEW

**Comment by apaszke** [comment]
Can you please revert the changes to Linear and Padding? We don't support non-batched mode anymore, so `nInputDims` makes no sense (it's always `== input.dim() - 1`).

**Comment by mvitez** [comment]
Then we need something in load_lua that converts networks to batched mode, but I am not sure that it can be always done automatically. Or should we leave people on their own to patch their networks?

**Comment by apaszke** [comment]
What kind of conversions are needed? I don't think there's anything special to do, is there?

**Comment by mvitez** [comment]
Without all my modifications here, several Eugenio's networks won't work. They have Padding.nInputDim=3 and Padding.index=1 (1-based Torch indexes), so Padding.index has to become 2, because the input is 4D instead of 3D. Then they have a View to 1D, followed by a Linear.

**Comment by culurciello** [comment]
All our nets were trained in batch mode, and even just having a small converter code will be enough, as long as it works
